packages/cactus-test-tooling/src/main/typescript/substrate-test-ledger/substrate-test-ledger.ts

```diff
@@ -0,0 +1,202 @@
+import type { EventEmitter } from "events";
+import { Optional } from "typescript-optional";
+import { RuntimeError } from "run-time-error";
+import type { Container, ContainerInfo } from "dockerode";
+import Docker from "dockerode";
+import { Logger, Checks, Bools } from "@hyperledger/cactus-common";
+import type { LogLevelDesc } from "@hyperledger/cactus-common";
+import { LoggerProvider } from "@hyperledger/cactus-common";
+import { Containers } from "../common/containers";
+
+export interface ISubstrateTestLedgerOptions {
+  readonly publishAllPorts: boolean;
+  readonly logLevel?: LogLevelDesc;
+  readonly imageName?: string;
+  readonly imageTag?: string;
+  readonly emitContainerLogs?: boolean;
+  readonly envVars?: Map<string, string>;
+}
+
+export class SubstrateTestLedger {
+  public static readonly CLASS_NAME = "SubstrateTestLedger";
+
+  public readonly logLevel: LogLevelDesc;
+  public readonly imageName: string;
+  public readonly imageTag: string;
+  public readonly imageFqn: string;
+  public readonly log: Logger;
+  public readonly emitContainerLogs: boolean;
+  public readonly publishAllPorts: boolean;
+  public readonly envVars: Map<string, string>;
+
+  private _containerId: Optional<string>;
+
+  public get containerId(): Optional<string> {
+    return this._containerId;
+  }
+
+  public get container(): Optional<Container> {
+    const docker = new Docker();
+    return this.containerId.isPresent()
+      ? Optional.ofNonNull(docker.getContainer(this.containerId.get()))
+      : Optional.empty();
+  }
+
+  public get className(): string {
+    return SubstrateTestLedger.CLASS_NAME;
+  }
+
+  constructor(public readonly opts: ISubstrateTestLedgerOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+
+    this.publishAllPorts = opts.publishAllPorts;
+    this._containerId = Optional.empty();
+    this.imageName =
+      opts.imageName || "ghcr.io/hyperledger/cactus-substrate-test-ledger";
+    this.imageTag = opts.imageTag || "0.0.1";
+    this.imageFqn = `${this.imageName}:${this.imageTag}`;
+    this.envVars = opts.envVars || new Map();
+    this.emitContainerLogs = Bools.isBooleanStrict(opts.emitContainerLogs)
+      ? (opts.emitContainerLogs as boolean)
+      : true;
+
+    this.logLevel = opts.logLevel || "INFO";
+
+    const level = this.logLevel;
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.log.debug(`Created instance of ${this.className} OK`);
+  }
+  public getContainerImageName(): string {
+    return `${this.imageName}:${this.imageTag}`;
+  }
+  public async start(omitPull = false): Promise<Container> {
+    const docker = new Docker();
+    if (this.containerId.isPresent()) {
+      this.log.warn(`Container ID provided. Will not start new one.`);
```

Comment:
@RafaelAPB @CatarinaPedreira If this is a feature then it should be logged on a level that is less severe than warning/error. A warning level error indicates that if you see that in the logs you should look into something *may* be a problem, but here it appears that if someone specified the container ID then that is one of the expected use-cases.

Reply:
You're right - I'm changing it to this.log.debug